libsolidity/interface/CompilerStack.cpp

```diff
@@ -1200,9 +1200,25 @@ void CompilerStack::resolveImports()
 		sourceOrder.push_back(_source);
 	};
 
+	vector<PragmaDirective const*> experimentalPragmaDirectives;
 	for (auto const& sourcePair: m_sources)
+	{
 		if (isRequestedSource(sourcePair.first))
 			toposort(&sourcePair.second);
+		if (sourcePair.second.ast && sourcePair.second.ast->experimentalSolidity())
+			for (ASTPointer<ASTNode> const& node: sourcePair.second.ast->nodes())
+				if (PragmaDirective const* pragma = dynamic_cast<PragmaDirective*>(node.get()))
+					if (pragma->literals().size() >=2 && pragma->literals()[0] == "experimental" && pragma->literals()[1] == "solidity")
+						experimentalPragmaDirectives.push_back(pragma);
```

Comment:
```suggestion
					if (pragma->literals().size() >=2 && pragma->literals()[0] == "experimental" && pragma->literals()[1] == "solidity")
					{
						experimentalPragmaDirectives.push_back(pragma);
						break;
					}
```

This runs before we make sure that there's only one of those pragmas in each source unit, isn't it?
So we need to stop looking through the nodes at the first one we find, right? Beyond that, there's no need to continue afterwards anyways.

Reply:
It'll probably work either way, since if you have more than one in a single file, that'll be an error later - but it's better to *properly* make sure here that we look at each file and only count one per file, since we may change the analysis paths based on this eventually.

Reply:
Yes, it'll work regardless, the duplicate checking is performed in the syntax analysis stage. I'll add the `break` statement, shouldn't really make much difference I think.